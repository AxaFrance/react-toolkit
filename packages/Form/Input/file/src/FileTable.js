import React from 'react';
import PropTypes from 'prop-types';
import { ClassManager } from '@axa-fr/react-toolkit-core';
import LineFile from './FileLine';

const propTypes = {
  values: PropTypes.array,
  errors: PropTypes.array,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  className: PropTypes.string,
  classModifier: PropTypes.string,
};
const defaultClassName = 'custom-table-file af-file-table';
const defaultProps = {
  values: null,
  errors: null,
  disabled: false,
  className: defaultClassName,
  classModifier: null,
};

const Errors = errors => {
  if (!errors || errors.length <= 0) {
    return null;
  }

  const errorList = errors.map(err =>
    err ? (
      <li key={err}>
        {err.file.name} ({err.file.size})
      </li>
    ) : (
      ''
    )
  );
  return (
    <div className="af-form__file-errors">
      <span>Le chargement de certains fichiers a échoué : </span>
      <ul className="af-form__file-errors-list">{errorList}</ul>
    </div>
  );
};

const FileTable = ({
  errors,
  values,
  className,
  classModifier,
  disabled,
  onClick,
}) => {
  if (!values || values.length <= 0) {
    return null;
  }
  const componentClassName = ClassManager.getComponentClassName(
    className,
    classModifier,
    defaultClassName
  );
  return (
    <div className={componentClassName}>
      {Errors(errors)}
      <ul className="af-form__file-list">
        {values.map(({ file, id }) => (
          <LineFile
            disabled={disabled}
            file={file}
            onClick={onClick}
            id={id}
            key={id}
          />
        ))}
      </ul>
    </div>
  );
};

FileTable.propTypes = propTypes;
FileTable.defaultProps = defaultProps;

export default FileTable;
