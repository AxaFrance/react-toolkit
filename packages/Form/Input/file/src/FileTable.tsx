import React, { ComponentProps, ComponentPropsWithoutRef } from 'react';
import { ClassManager } from '@axa-fr/react-toolkit-core';
import LineFile from './FileLine';

const defaultClassName = 'custom-table-file af-file-table';

const Errors = ({ errors }: Pick<Props, 'errors'>) => {
  return (
    <div className="af-form__file-errors">
      <span>Le chargement de certains fichiers a échoué : </span>
      <ul className="af-form__file-errors-list">
        {errors.map((err) => (
          <li key={err.file.name}>
            {err.file.name} ({err.file.size})
          </li>
        ))}
      </ul>
    </div>
  );
};

type FileLineAction = Pick<ComponentProps<typeof LineFile>, 'onClick'>;
type File = Pick<ComponentPropsWithoutRef<typeof LineFile>, 'file'>;
type Props = FileLineAction & {
  errors?: File[];
  values?: (File & { id: string })[];
  className?: string;
  classModifier?: string;
  disabled?: boolean;
};
const FileTable = ({
  errors,
  values,
  className = defaultClassName,
  classModifier,
  disabled,
  onClick,
}: Props) => {
  const componentClassName = ClassManager.getComponentClassName(
    className,
    classModifier,
    defaultClassName
  );
  return (
    <div className={componentClassName}>
      {errors && errors.length > 0 && <Errors errors={errors} />}
      {values && values.length > 0 && (
        <ul className="af-form__file-list">
          {values.map(({ file, id }) => (
            <LineFile
              disabled={disabled}
              file={file}
              onClick={onClick}
              id={id}
              key={id}
            />
          ))}
        </ul>
      )}
    </div>
  );
};

export default FileTable;
