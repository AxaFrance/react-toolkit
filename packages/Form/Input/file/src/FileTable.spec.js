import React from 'react';
import { shallow } from 'enzyme';
import FileTable, { Errors } from './FileTable';

describe('<File.FileTable>', () => {
  it('renders File.FileTable correctly', () => {
    const wrapper = shallow(
      <FileTable
        errors={[{ file: { name: 'youhou.txt', size: 2 } }]}
        values={[{ file: {}, id: 'id' }]}
        onClick={selectedId => console.log(selectedId)}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders File.FileTable correctly when no have values', () => {
    const wrapper = shallow(
      <FileTable
        errors={[{ file: { name: 'youhou.txt', size: 2 } }]}
        values={[]}
        onClick={selectedId => console.log(selectedId)}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Errors correctly with errors', () => {
    const wrapper = shallow(
      <Errors errors={[{ file: { name: 'youhou.txt', size: 2 } }]} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders Errors correctly with empty errors', () => {
    const wrapper = shallow(<Errors errors={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
