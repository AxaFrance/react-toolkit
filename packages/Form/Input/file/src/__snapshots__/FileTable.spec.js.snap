// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<File.FileTable> renders Errors correctly with empty errors 1`] = `""`;

exports[`<File.FileTable> renders Errors correctly with errors 1`] = `
<div
  className="af-form__file-errors"
>
  <span>
    Le chargement de certains fichiers a échoué : 
  </span>
  <ul
    className="af-form__file-errors-list"
  >
    <li
      key="[object Object]"
    >
      youhou.txt
       (
      2
      )
    </li>
  </ul>
</div>
`;

exports[`<File.FileTable> renders File.FileTable correctly 1`] = `
<div
  className="custom-table-file af-file-table"
>
  <Errors
    errors={
      Array [
        Object {
          "file": Object {
            "name": "youhou.txt",
            "size": 2,
          },
        },
      ]
    }
  />
  <ul
    className="af-form__file-list"
  >
    <FileLine
      classModifier={null}
      className="af-form__file-line"
      disabled={false}
      file={Object {}}
      id="id"
      key="id"
      onClick={[Function]}
    />
  </ul>
</div>
`;

exports[`<File.FileTable> renders File.FileTable correctly when no have values 1`] = `""`;
